tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -0,0 +1,93 @@
+// <copyright file="DiscoveryService.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Agent.DiscoveryService;
+
+internal class DiscoveryService : IDiscoveryService
+{
+    private static readonly string[] SupportedDebuggerEndpoints = new[] { "debugger/v1/input" };
+    private static readonly string[] SupportedProbeConfigurationEndpoints = new[] { "v0.7/config" };
+
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DiscoveryService>();
+
+    private readonly IApiRequestFactory _apiRequestFactory;
+    private readonly Uri _agentUri;
+
+    private DiscoveryService(
+        Uri agentUri,
+        IApiRequestFactory apiRequestFactory)
+    {
+        _agentUri = agentUri;
+        _apiRequestFactory = apiRequestFactory;
+    }
+
+    public static string[] AllSupportedEndpoints => SupportedDebuggerEndpoints.Concat(SupportedProbeConfigurationEndpoints).ToArray();
+
+    public string ProbeConfigurationEndpoint { get; private set; }
+
+    public string DebuggerEndpoint { get; private set; }
+
+    public string AgentVersion { get; private set; }
+
+    public static DiscoveryService Create(IConfigurationSource configurationSource, IApiRequestFactory apiRequestFactory)
+    {
+        var exporterSettings = new ExporterSettings(configurationSource);
+        return new DiscoveryService(exporterSettings.AgentUri, apiRequestFactory);
+    }
```

Comment:
This is a "dangerous" assumption to make as some customers update the exporter settings in code, and on top of that, the `AgentUri` might not be an Http URI (e.g. UDS/named pipes). Might be worth looking at how the `TelemetryController` handles it?

Reply:
Yes, I agree with you, but for now we are calling `Discover` only from inside of `LiveDebugger`, and as mentioned in above comments, I've disabled initialization for UDS/Named pipes.
I think we should:
1. Move `DiscoveryService` outside of `LiveDebugger` initialization, as other products will use it, too.
2. Solve UDS/named pipes issue, of course. As mentioned before, we will fix it in near future.